packages/fx-core/tests/component/generator/generator.test.ts

```diff
@@ -456,20 +476,31 @@ describe("Generator happy path", async () => {
     }
   });
 
-  /*
-  TODO: fix the wrong test
+  // TODO: fix the wrong test
   it("external sample", async () => {
     const axiosStub = sandbox.stub(axios, "get");
-    const sampleName = "bot-proactive-messaging-teamsfx";
+    sandbox.stub(sampleProvider, "SampleCollection").value(mockedExternalSampleConfig);
+
+    const sampleName = "test";
     const mockFileName = "test.txt";
     const mockFileData = "test data";
-    const fileInfo = [{ type: "file", path: `${sampleName}/${mockFileName}` }];
-    axiosStub.onFirstCall().resolves({ status: 200, data: { tree: fileInfo } });
+    const foobarName = "foobar";
+    const foobarFileName = "foobar.txt";
+    const fileInfo = [
+      { type: "file", path: `sample/${sampleName}/${mockFileName}` },
+      { type: "file", path: `sample/${foobarName}/${mockFileName}` },
+    ];
+    axiosStub.onFirstCall().resolves({ status: 200, data: { tree: fileInfo } })
     axiosStub.onSecondCall().resolves({ status: 200, data: mockFileData });
     const result = await Generator.generateSample(context, tmpDir, sampleName);
+    if (!fs.existsSync(tmpDir + "/" + mockFileName)) {
```

Comment:
The path delimiter varies between operating systems. It would be better to use [path.join](https://nodejs.org/api/path.html#pathjoinpaths).